CredentialsManager/Models/Credentials.cs

```diff
@@ -0,0 +1,100 @@
+﻿using System;
+
+namespace CredentialsManager.Models
+{
+  public class Credentials : IEquatable<Credentials>, IComparable<Credentials>, IComparable
+  {
+    public string UserName { get; }
+    public string UserPassword { get; }
+
+    public Credentials(string userName, string userPassword)
+    {
+      if (string.IsNullOrWhiteSpace(userName))
+      {
+        throw new ArgumentException("Value can not be null or empty", nameof(userName));
+      }
+
+      if (string.IsNullOrWhiteSpace(userPassword))
+      {
+        throw new ArgumentException("Value can not be null or empty", nameof(userName));
+      }
+      
+      UserName = userName;
+      UserPassword = userPassword;
+    }
+
+    public static bool TryParse(string line, string delimiter, out Credentials credentials)
+    {
+      credentials = default;
+      const int validRowLength = 2;
+
+      if (string.IsNullOrWhiteSpace(line)) return false;
+
+      var row = line.Split(delimiter);
+      if (row.Length != validRowLength) return false;
+
+      var userName = row[0].Trim();
+      var userPassword = row[1].Trim();
+
+      if (string.IsNullOrWhiteSpace(userName) || string.IsNullOrWhiteSpace(userPassword))
+      {
+        return false;
+      }
+
+      credentials = new Credentials(userName, userPassword);
+
+      return true;
+    }
+
+    public override string ToString()
+    {
+      return $"{UserName},{UserPassword}";
```

Comment:
This is helpful for debugging and this is just a toy homework application so no big deal.  However, in a real world application, you always have to be aware of how your password is being treated.  For example, this might accidentally lead to a password getting leaked by some debugger or exception or accidental print.

Reply:
I will definitely keep this in mind. Thank you.